File: Moreno_java_Herencia/Cuenta/CuentaCorriente.java
package Cuenta;

public class CuentaCorriente extends Cuenta {
    private double limiteSobregiro;

    public CuentaCorriente(String titular, int numeroCuenta, double saldo, double limiteSobregiro){
        super(titular, numeroCuenta, saldo);
        this.limiteSobregiro=limiteSobregiro;
    }

    @Override
    public void retirar(double cantidad){
        if (cantidad <=0) {
            System.out.println("El valor a retirar debe ser mayor a 0");
            return;
        }

        double saldoDisponible = getSaldo()+limiteSobregiro;
        if (cantidad > saldoDisponible) {
            System.out.println("Fondos insuficientes. limite alcanzado. maximo disponible: "+saldoDisponible);
            return;
        }

        setSaldo(getSaldo()-cantidad);
        System.out.println("Retiro exitoso: "+cantidad+" nuevo saldo: "+getSaldo());
    }

    @Override
    public void mostrarInformacion(){
        super.mostrarInformacion();
        System.out.println("tipo: Cuenta Corriente ");
        System.out.println("limite de sobregiro: "+limiteSobregiro);
        System.out.println("Saldo disponible: "+ (getSaldo()+limiteSobregiro));
    }

    public double getLimiteSobreGiro(){
        return limiteSobregiro;
    }

}

File: Banco/Cuenta_Bancaria/CuentaCorriente.ts
import { CuentaBancaria } from "./CuentaBancaria";

export class CuentaCorriente extends CuentaBancaria {
    private _limiteSobregiro: number;

    constructor(titular: string, numeroCuenta: string, saldoInicial: number = 0, limiteSobregiro: number = 500) {
        super(titular, numeroCuenta, saldoInicial);
        this._limiteSobregiro = limiteSobregiro;
    }

    retirar(cantidad: number): void {
        if (cantidad <= 0) {
            console.log("El monto a retirar debe ser mayor a cero.");
            return;
        }

        const saldoDisponible = this.getSaldo + this._limiteSobregiro;
        
        if (cantidad > saldoDisponible) {
            console.log(`Fondos insuficientes. Límite de sobregiro alcanzado. Máximo disponible: ${saldoDisponible.toFixed(2)}`);
            return;
        }

        this.setSaldo = this.getSaldo - cantidad;
        console.log(`Retiro exitoso: ${cantidad}. Nuevo saldo: ${this.getSaldo}`);
    }

    mostrarDatos(): void {
        super.mostrarDatos();
        console.log(`Tipo: Cuenta Corriente`);
        console.log(`Límite de Sobregiro: ${this._limiteSobregiro}`);
        console.log(`Saldo Disponible: ${(this.getSaldo + this._limiteSobregiro)}`);
    }

    public get getLimiteSobregiro(): number {
        return this._limiteSobregiro;
    }
}